olimex/teres_i/default.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ...}:
+
+{
+  hardware.deviceTree = {
+    name = lib.mkDefault "allwinner/sun50i-a64-teres-i.dts";
+    enable = lib.mkDefault true;
```

Comment:
This is not needed.

 - https://github.com/NixOS/nixpkgs/blob/81e8f48ebdecf07aab321182011b067aafc78896/nixos/modules/hardware/device-tree.nix#L109-L110

Reply:
> This is not needed.
> 
> * https://github.com/NixOS/nixpkgs/blob/81e8f48ebdecf07aab321182011b067aafc78896/nixos/modules/hardware/device-tree.nix#L109-L110 -- @samueldr (https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1349258535)

in my tests it was sometimes using the [sun50i-a64.dtsi](https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/allwinner/sun50i-a64.dtsi) so this is an attempt to mitigate that behavior (also observed on armbian).

It doesn't seem to be using the DTB if `hardware.deviceTree.enable` is set to zero.

Additionally i tested other kernels and they don't seem to work correctly if this is set so expecting the user to use the mainline, ideally it should have an option for linux-libre, but that seems to need an upstream changes to work correctly.

Reply:
> It doesn't seem to be using the DTB if `hardware.deviceTree.enable` is set to zero.

It simply can't. It's either `true` or  `false`, and on `aarch64-linux` `pkgs.stdenv.hostPlatform.linux-kernel.DTB` is always true, since it's a platform value.

So `hardware.deviceTree.enable` is ***always*** true for AArch64.

Reply:
misspoke about the zero meant `false`

I do more tests on this and get back to you